osmdroid-android/src/main/java/org/osmdroid/views/overlay/PolyOverlayWithIW.java

```diff
@@ -0,0 +1,233 @@
+package org.osmdroid.views.overlay;
+
+import android.graphics.Canvas;
+import android.graphics.Paint;
+import android.graphics.Path;
+
+import org.osmdroid.util.GeoPoint;
+import org.osmdroid.util.PointL;
+import org.osmdroid.views.MapView;
+import org.osmdroid.views.Projection;
+import org.osmdroid.views.overlay.infowindow.InfoWindow;
+import org.osmdroid.views.overlay.milestones.MilestoneManager;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Repository of common methods for Polyline and Polygon
+ * @author Fabrice Fontaine
+ * @since 6.1.0
+ */
+public abstract class PolyOverlayWithIW extends OverlayWithIW {
+
+	protected LinearRing mOutline;
+	protected List<LinearRing> mHoles = new ArrayList<>();
+	protected final Paint mOutlinePaint = new Paint();
+	protected Paint mFillPaint;
+	private List<MilestoneManager> mMilestoneManagers = new ArrayList<>();
+	private GeoPoint mInfoWindowLocation;
+
+	private final LineDrawer mLineDrawer;
+	protected final Path mPath;
+	protected float mDensity = 1.0f;
+	protected List<GeoPoint> mOriginalPoints = new ArrayList<>();
+
+	protected PolyOverlayWithIW(final MapView pMapView, final boolean pUsePath, final boolean pClosePath) {
+		super();
+		if (pMapView != null) {
+			setInfoWindow(pMapView.getRepository().getDefaultPolylineInfoWindow());
+			mDensity = pMapView.getContext().getResources().getDisplayMetrics().density;
+		}
+		if (pUsePath) {
+			mPath = new Path();
+			mLineDrawer = null;
+			mOutline = new LinearRing(mPath, pClosePath);
+		} else {
+			mPath = null;
+			mLineDrawer = new LineDrawer(256);
+			mOutline = new LinearRing(mLineDrawer);
+			////mOutline.clearPath();
```

Comment:
I think this comment could be removed?

Reply:
Fair enough, `mOutline.clearPath();` should be executed and not commented.